### PR TITLE
Adds interactivity when generating provider documentation.

### PR DESCRIPTION
**Pull request by potiuk**
The release manager, when reviewing providers to release might
make interactive decisions what to do:

1) mark certain provider as 'doc-only' change
2) decide whethere to generate documentation for the provider

In case the provider change is marked as 'doc-only' the next time
when providers are checked the doc-only change is not seen as
'change' and the provider is automatically skipped.

This saves time when preparing subsequent releases of providers
as all the "doc-only changes" from the previous release do not
have to be re-reviewed (unless there are some new changes).

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

**Comment by potiuk** [comment]
Hey @kaxil @ash - I'd love to release those few providers for PIP21 compatibility - this would be useful to have for that.

File: dev/provider_packages/prepare_provider_packages.py [hunk 1]
```diff
@@ -125,21 +125,20 @@ def cli():
...
-
 option_git_update = click.option(
     '--git-update/--no-git-update',
     default=True,
     is_flag=True,
     help=f"If the git remote {HTTPS_REMOTE} already exists, don't try to update it",
 )
+
+option_interactive = click.option(
+    '--interactive/--no-interactive',
```
Comment (kaxil) [inline dev/provider_packages/prepare_provider_packages.py]:
```suggestion
    '--interactive/--non-interactive',
```

Worth changing `--no-interactive` to `non-interactive` everywhere

File: dev/provider_packages/prepare_provider_packages.py [hunk 2]
```diff
@@ -791,36 +804,38 @@ def convert_git_changes_to_table(
     :param changes: list of changes in a form of multiple-line string
     :param base_url: base url for the commit URL
     :param markdown: if True, markdown format is used else rst
-    :return: formatted table
+    :return: formatted table + list of changes (starting from the latest
```
Comment (kaxil) [inline dev/provider_packages/prepare_provider_packages.py]: Missing close bracket

Comment (kaxil) [inline dev/provider_packages/prepare_provider_packages.py]:
```suggestion
    :return: formatted table + list of changes (starting from the latest)
```

File: dev/provider_packages/prepare_provider_packages.py (hunk 1 above)
Comment (potiuk) [reply dev/provider_packages/prepare_provider_packages.py]: I am afraid this is not how click works :( https://click.palletsprojects.com/en/7.x/options/#boolean-flags

Comment (potiuk) [reply dev/provider_packages/prepare_provider_packages.py]: Ah yeah Indeed.

**Comment by potiuk** [comment]
All Resolved

**Review by kaxil — APPROVED** [review]

**Pull request merged**